synapse/storage/databases/main/stream.py

```diff
@@ -895,7 +909,7 @@ def _reset_federation_positions_txn(self, txn) -> None:
             GROUP BY type
         """
         txn.execute(sql)
-        min_positions = dict(txn)  # Map from type -> min position
+        min_positions = {typ: pos for typ, pos in txn}  # Map from type -> min position
```

Comment:
Was this to just let mypy figure out the type?

Reply:
Oh, this is because mypy complains that `txn` implements `Iterable[Tuple[Any, ...]]` rather than `Iterable[Tuple[Any, Any]]`, whereas with this form we're (implicitly) explicitly asserting that the tuple length is two. I think. But yes its to appease mypy